File: simulation/0038-Count-and-Say/main.go
package main

import (
	"strconv"
	"strings"
)

func countAndSay(n int) string {
	res := "1"
	for i := 1; i < n; i++ {
		var buf strings.Builder
		count := 0
		var now byte
		for j := 0; j < len(res); j++ {
			if j == 0 {
				now = res[j]
				count = 1
			} else {
				if now == res[j] {
					count++
				} else {
					buf.WriteString(strconv.Itoa(count))
					buf.WriteString(string(now))
					now = res[j]
					count = 1
				}
			}
		}
		buf.WriteString(strconv.Itoa(count))
		buf.WriteString(string(now))
		res = buf.String()
	}
	return res
}

File: simulation/0038-Count-and-Say/main.py
class Solution:
    def countAndSay(self, n: int) -> str:
        res = "1"
        for i in range(1, n):
            buf = []
            count = 1
            now = res[0]
            for j in range(1, len(res)):
                if now == res[j]:
                    count += 1
                else:
                    buf.append(str(count))
                    buf.append(now)
                    now = res[j]
                    count = 1
            buf.append(str(count))
            buf.append(now)

            res = "".join(buf)

        return res
